api/client/client.go

```diff
@@ -1319,6 +1319,61 @@ func (c *Client) DeleteAllNodes(ctx context.Context, namespace string) error {
 	return trail.FromGRPC(err)
 }
 
+// StreamSessionEvents streams audit events from a given session recording.
+func (c *Client) StreamSessionEvents(ctx context.Context, sessionID string, startIndex int) (chan events.AuditEvent, chan error) {
+	request := &proto.StreamSessionEventsRequest{
+		SessionID:  sessionID,
+		StartIndex: int32(startIndex),
+	}
+
+	ch := make(chan events.AuditEvent)
+
+	stream, err := c.grpc.StreamSessionEvents(ctx, request)
+	if err != nil {
+		e := make(chan error)
+		go func() {
+			select {
+			case e <- trace.Wrap(err):
+			case <-ctx.Done():
+			}
+			close(ch)
+		}()
+
+		return ch, e
+	}
```

Comment:
Related to sasha's comment, my one suggestion at this point is to change the "contract" of the API so that the event channel will *not* be closed if there is an error. This eliminates the need to worry if there will be a race between the caller getting the error or seeing the closed channel, and you can just write the error to a buffered channel and return rather than creating a goroutine and worrying whether it will leak.

Sorry to suggest so many changes on this (you can take or leave any of them). I think I mentioned this idea and the current one earlier but now that we are considering that this goroutine could leak I really think it would be simpler.

Reply:
Yeah, fair enough. I'll make this change since it does seem beneficial. I'm still not too accustomed to deal with so many channels and the various patterns for it.

Reply:
@nklaassen @klizhentas Changed the contract to not close c on error conditions and removed the goroutines used to pushing to the err channel since they are now buffered.